cmd/mailserver-canary/main.go

```diff
@@ -0,0 +1,316 @@
+// mailserver-canary tests whether a mailserver enode responds to a historic messages request.
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	stdlog "log"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto/sha3"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/p2p/discv5"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+	"github.com/status-im/status-go/api"
+	"github.com/status-im/status-go/logutils"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/rpc"
+	"github.com/status-im/status-go/services/shhext"
+	"github.com/status-im/status-go/t/helpers"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+const (
+	mailboxPassword = "status-offline-inbox"
+)
+
+// All general log messages in this package should be routed through this logger.
+var logger = log.New("package", "status-go/cmd/mailserver-canary")
+
+var (
+	enodeAddr        = flag.String("mailserver", "", "mailserver enode address (e.g. enode://1da276e34126e93babf24ec88aac1a7602b4cbb2e11b0961d0ab5e989ca9c261aa7f7c1c85f15550a5f1e5a5ca2305b53b9280cf5894d5ecf7d257b173136d40@167.99.209.61:30504)")
+	publicChannel    = flag.String("channel", "status", "The public channel name to retrieve historic messages from")
+	period           = flag.Int("period", 24*60*60, "How far in the past to request messages from mailserver, in seconds")
+	minPow           = flag.Float64("shh.pow", params.WhisperMinimumPoW, "PoW for messages to be added to queue, in float format")
+	ttl              = flag.Int("shh.ttl", params.WhisperTTL, "Time to live for messages, in seconds")
+	homePath         = flag.String("home-dir", ".", "Home directory where state is stored")
+	logLevel         = flag.String("log", "INFO", `Log level, one of: "ERROR", "WARN", "INFO", "DEBUG", and "TRACE"`)
+	logFile          = flag.String("logfile", "", "Path to the log file")
+	logWithoutColors = flag.Bool("log-without-color", false, "Disables log colors")
+)
+
+func main() {
+	initialize()
+
+	if enodeAddr == nil {
+		logger.Crit("No mailserver address specified", "enodeAddr", *enodeAddr)
+		os.Exit(1)
+	}
+
+	mailserverParsedNode, err := discv5.ParseNode(*enodeAddr)
+	if err != nil {
+		logger.Crit("Invalid mailserver address specified", "enodeAddr", *enodeAddr, "error", err)
+		os.Exit(1)
+	}
+
+	verifyMailserverBehavior(mailserverParsedNode)
+	logger.Info("Mailserver responded correctly", "address", enodeAddr)
+	os.Exit(0)
+}
+
+func initialize() {
```

Comment:
In Go there is a function called `init()` which is automatically called before `main()`.